lib/routes/emails.js

```diff
@@ -470,6 +483,16 @@ module.exports = (log, db, mailer, config, customs, push) => {
             () => reply({}),
             reply
           )
+
+        function isServerVerified(code, uid, requestSignature) {
+          const signature = crypto.createHmac('sha256', SERVER_VERIFICATION_KEY)
+          .update([
+            code,
+            uid
+          ].join('\n'))
+          .digest('hex')
+          return signature === requestSignature
```

Comment:
We'll need to use a constant-time string comparison here to avoid timing attacks [1], see for example the way we do it when checking flow signatures:

  https://github.com/mozilla/fxa-auth-server/blob/master/lib/metrics/context.js#L173

[1] https://codahale.com/a-lesson-in-timing-attacks/